x-pack/plugins/uptime/public/hooks/use_route_change.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useEffect } from 'react';
+import { useRouteMatch } from 'react-router-dom';
+import {
+  CERTIFICATES_ROUTE,
+  OVERVIEW_ROUTE,
+  SETTINGS_ROUTE,
+  MONITOR_ROUTE,
+} from '../../common/constants';
+
+export const useRouteChange = () => {
+  const certPage = useRouteMatch(CERTIFICATES_ROUTE);
+  const settingsPage = useRouteMatch(SETTINGS_ROUTE);
+  const overviewPage = useRouteMatch(OVERVIEW_ROUTE);
+  const monitorPage = useRouteMatch(MONITOR_ROUTE);
+
+  useEffect(() => {
```

Comment:
This feels really fragile. If we add a new page we have to remember to update this page. We probably won't. 

Make we should make this a part of the page header, and use a prop passed in there instead.

Reply:
i refactored it, use typescript now it will be enforced and hopefully next next page addition will be taken care of.